src/main/java/org/arkecosystem/crypto/transactions/Deserializer.java

```diff
@@ -25,17 +25,23 @@ public Transaction deserialize(String serialized) {
 
         int assetOffset = deserializeHeader();
         deserializeTypeSpecific(assetOffset);
+
         deserializeVersionOne();
 
         return this.transaction;
     }
 
-    public int deserializeHeader() {
+    private int deserializeHeader() {
         transaction.version = this.buffer.get();
         transaction.network = this.buffer.get();
-        transaction.type = Types.values()[this.buffer.get()];
-        transaction.timestamp = this.buffer.getInt();
-
+        if (transaction.version == 1) {
+            transaction.type = TransactionType.values()[this.buffer.get()];
+            transaction.timestamp = this.buffer.getInt();
+        }else {
+            transaction.typeGroup = TransactionTypeGroup.values()[this.buffer.getInt()];
+            transaction.type = TransactionType.values()[this.buffer.getShort()];
```

Comment:
Not entirely sure, but doesn't this potentially cause issues when `type == 0` with `typeGroup != 1` is picked as a `Transfer` ? If Java handles it fine, nvm me.